go/cli/cmd/kagent/main.go

```diff
@@ -209,7 +210,166 @@ func main() {
+If no model is specified, the agent will need to be configured later.
+
+Examples:
+  kagent init adk python dice
+  kagent init adk python dice --instruction-file instructions.md
+  kagent init adk python dice --model-provider Gemini --model-name gemini-2.0-flash`,
+		Args: cobra.ExactArgs(3),
+		Run: func(cmd *cobra.Command, args []string) {
+			initCfg := &cli.InitCfg{
+				Config:    cfg,
+				Framework: args[0],
+				Language:  args[1],
+				AgentName: args[2],
+			}
+
+			// Get instruction file path if specified
+			if instructionFile, _ := cmd.Flags().GetString("instruction-file"); instructionFile != "" {
+				initCfg.InstructionFile = instructionFile
+			}
+
+			// Get model provider and name if specified
+			if modelProvider, _ := cmd.Flags().GetString("model-provider"); modelProvider != "" {
+				initCfg.ModelProvider = modelProvider
+			}
+
+			if modelName, _ := cmd.Flags().GetString("model-name"); modelName != "" {
+				initCfg.ModelName = modelName
+			}
+
+			if description, _ := cmd.Flags().GetString("description"); description != "" {
+				initCfg.Description = description
+			}
+
+			if err := cli.InitCmd(initCfg); err != nil {
+				fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+				os.Exit(1)
+			}
+		},
+		Example: `kagent init adk python dice`,
+	}
+
+	// Add flags for custom instructions and model selection
+	initCmd.Flags().String("instruction-file", "", "Path to file containing custom instructions for the root agent")
+	initCmd.Flags().String("model-provider", "", "Model provider (OpenAI, Anthropic, AzureOpenAI, Gemini)")
+	initCmd.Flags().String("model-name", "", "Model name (e.g., gpt-4, claude-3-5-sonnet, gemini-2.0-flash)")
+	initCmd.Flags().String("description", "", "Description for the agent")
+
+	buildCmd := &cobra.Command{
+		Use:   "build [project-directory]",
+		Short: "Build a Docker image for an agent project",
+		Long: `Build a Docker image for an agent project created with the init command.
+
+This command will look for a Dockerfile in the specified project directory and build
+a Docker image using docker build. The image can optionally be pushed to a registry.
+
+Image naming:
+- If --image is provided, it will be used as the full image specification (e.g., ghcr.io/myorg/my-agent:v1.0.0)
+- Otherwise, defaults to localhost:5001/{agentName}:latest where agentName is loaded from kagent.yaml
+
+Examples:
+  kagent build ./my-agent
+  kagent build ./my-agent --image ghcr.io/myorg/my-agent:v1.0.0
+  kagent build ./my-agent --image ghcr.io/myorg/my-agent:v1.0.0 --push`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			buildCfg := &cli.BuildCfg{
+				Config: cfg,
+			}
+
+			// Get project directory from positional argument
+			buildCfg.ProjectDir = args[0]
+
+			if image, _ := cmd.Flags().GetString("image"); image != "" {
+				buildCfg.Image = image
+			}
+
+			if push, _ := cmd.Flags().GetBool("push"); push {
+				buildCfg.Push = true
+			}
```

Comment:
Why are you doing `gets` directly rather than setting these values on a config struct?

Reply:
updated these flags to bind to the config struct instead.